src/realm/object-store/sync/impl/sync_metadata.cpp

```diff
@@ -182,7 +182,19 @@ SyncMetadataManager::SyncMetadataManager(std::string path, bool should_encrypt,
         }
     };
 
-    SharedRealm realm = Realm::get_shared_realm(config);
+    m_metadata_config = std::move(config);
+
+    SharedRealm realm;
+    try {
+        realm = get_realm();
+    }
+    catch (const RealmFileException&) {
+        if (!should_encrypt)
```

Comment:
This should check the error code from the exception rather than should_encrypt.

Reply:
Opening with the wrong key and opening when should_encrypt don't match are both `RealmFileException::Kind::AccessError` and have the same error message. What other ways are there to differentiate?

Reply:
In both of those cases we want to delete the Realm and recreate it. Currently this code will delete the Realm when any error occurs if we want to encrypt the Realm even if it's an unrelated error, and will fail to delete the Realm in a case where we want to (disabling encryption when the existing file is encrypted).

Reply:
I think I understand. I previously was trying to retain the response of throwing an error when `should_encrypt == false` was applied to a path that was encrypted.